packages/cozy-procedures/src/redux/documentsDataSlice.js

```diff
@@ -104,45 +128,60 @@ export const {
   fetchDocumentError,
   setProcedureStatus,
   unlinkDocument,
-  linkDocumentSuccess
+  linkDocumentSuccess,
+  setLoadingFalse
 } = actions
 
-export function fetchDocument(client, documentTemplate) {
+export function fetchDocumentsByCategory(documentTemplate) {
   return async dispatch => {
-    dispatch(
-      fetchDocumentLoading({
-        loading: true,
-        idDoctemplate: documentTemplate
-      })
-    )
     try {
       const docWithRules = creditApplicationTemplate.documents[documentTemplate]
+      for (let i = 0; i < docWithRules.count; i++) {
+        dispatch(
+          fetchDocumentLoading({
+            idDoctemplate: documentTemplate,
+            index: i
+          })
+        )
+      }
 
       const files = await AdministrativeProcedure.getFilesByRules(docWithRules)
 
       if (files.data) {
+        files.data.map((file, index) => {
+          dispatch(
+            fetchDocumentSuccess({
+              file: file,
+              idDoctemplate: documentTemplate,
+              index
+            })
+          )
+        })
+      }
+
+      if (files.data.length < docWithRules.count) {
```

Comment:
It might be good to add a comment for this section, it took me a little bit to understand that we are unsetting the loading status on document holders for which we couldn't find a file.

Edit: Or maybe we can rename the action to something less generic?